tests/xtro-sharpie/SimdCheck.cs

```diff
@@ -0,0 +1,382 @@
+			// results.
+			{ "GKTriangle", new NativeSimdInfo { Managed = "GKTriangle", }},
+			// This is a 'vector_int4, represented by a Vector4i in managed code,
+			// which means it's matching the native definition.
+			{ "MDLVoxelIndex", new NativeSimdInfo { Managed = "MDLVoxelIndex" }},
+			// In managed code this is struct of two Vector4, so it's matching the native definition.
+			{ "MDLVoxelIndexExtent", new NativeSimdInfo { Managed = "MDLVoxelIndexExtent" }},
+			// In managed code this is a struct of two Vector3, so it's *not* matching
+			// the native definition. However, since we're manually marshalling this type
+			// (using [MarshalDirective]), managed code doesn't get incorrect results.
+			{ "MDLAxisAlignedBoundingBox", new NativeSimdInfo { Managed = "MDLAxisAlignedBoundingBox", }},
+			// The managed definition is identical to the native definition
+			{ "MPSImageHistogramInfo", new NativeSimdInfo { Managed = "MPSImageHistogramInfo" }},
+		};
+
+		static Dictionary<string, bool> managed_simd_types; // bool: invalid_for_simd
+
+		static SimdCheck ()
+		{
+			managed_simd_types = new Dictionary<string, bool> ();
+			foreach (var kvp in type_mapping) {
+				managed_simd_types [kvp.Value.Managed] = false;
+				if (!string.IsNullOrEmpty (kvp.Value.InvalidManaged))
+					managed_simd_types [kvp.Value.InvalidManaged] = true;
+			}
+		}
+
+		class ManagedSimdInfo
+		{
+			public MethodDefinition Method;
+			public bool ContainsInvalidMappingForSimd;
+		}
+		Dictionary<string, ManagedSimdInfo> managed_methods = new Dictionary<string, ManagedSimdInfo> ();
+
+		public override void VisitManagedMethod (MethodDefinition method)
+		{
+			var type = method.DeclaringType;
+
+			if (!type.IsNested && type.IsNotPublic)
+				return;
+
+			if (type.IsNested && (type.IsNestedPrivate || type.IsNestedAssembly || type.IsNestedFamilyAndAssembly))
+				return;
+
+			if (method.IsPrivate || method.IsAssembly || method.IsFamilyAndAssembly)
+				return; // Don't care about non-visible types
+
+			if (type.Namespace == "Simd" || type.Namespace.StartsWith ("OpenTK", StringComparison.Ordinal))
+				return; // We're assuming everything in the Simd and OpenTK namespaces can be ignored (the former because it's correctly written, the latter because it doesn't map to native simd types).
+
+			if (method.HasCustomAttributes && method.CustomAttributes.Where ((v) => v.Constructor.DeclaringType.Name == "ExtensionAttribute").Any ())
+				return; // Extension methods can't be mapped.
+
+			var invalid_simd_type = false;
+			var contains_simd_types = ContainsSimdTypes (method, ref invalid_simd_type);
+
+			var key = method.GetName ();
+			if (key == null) {
+				if (method.IsObsolete ())
+					return; // Don't care about obsolete API.
+
+				if (contains_simd_types && very_strict) {
+					// We can't map this method to a native function.
+					Console.WriteLine ($"!simd-can't-map-managed! {method}");
```

Comment:
start with either `!unknown-` or `!wrong-` so it gets sorted along other important/critical logs

Reply:
This is not important/critical, so I didn't change it.